tests/test_cli.py

```diff
@@ -406,22 +725,130 @@ def test_best_trials_command() -> None:
             storage_url,
             "--study-name",
             study_name,
-            "--format",
-            "json",
         ]
 
+        if output_format is not None:
+            command += ["--format", output_format]
+
         output = str(subprocess.check_output(command).decode().strip())
-        trials = json.loads(output)
+        trials = _parse_output(output, output_format or "table")
         best_trials = [trial.number for trial in study.best_trials]
 
         assert len(trials) == len(best_trials)
 
+        df = study.trials_dataframe(attrs, multi_index=True)
+
         for trial in trials:
-            assert set(trial.keys()) == set(df.columns)
-            assert trial["number"] in best_trials
-            for key, value in trial.items():
-                expected_value = df.loc[trial["number"]][key]
-                if isinstance(value, int) or isinstance(value, float):
+            if output_format is None or output_format == "table":
+                assert int(trial["number"]) in best_trials
+            else:
+                assert trial["number"] in best_trials
+            for key in df.columns:
+                if output_format is None or output_format == "table":
+                    expected_value = df.loc[int(trial["number"])][key]
+                else:
+                    expected_value = df.loc[trial["number"]][key]
```

Comment:
In this test function, we can see many branches that switch the assertions depending on `output_format`.
I think it is inevitable to share the test code, but I also think we can reduce the number of branches.
For example, can we change the code as follows:
 
```suggestion
            number = int(trial["number"]) if output_format in (None, "table") else trial["number"]
            assert number in best_trials
            for key in df.columns:
                expected_value = df.loc[number][key]
```

`output_format in (None, "table")` is a bit casual compared with `output_format is None or output_format == "table"`, but linter seems to accept it.